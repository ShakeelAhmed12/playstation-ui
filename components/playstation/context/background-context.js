import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const BackgroundProvider = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}