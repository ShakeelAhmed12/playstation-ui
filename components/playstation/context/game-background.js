import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const GameBackground = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}