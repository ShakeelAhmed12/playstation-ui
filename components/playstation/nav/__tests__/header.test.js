import React from 'react';
import renderer from 'react-test-renderer';
import { GameBackground } from '../../context/game-background';

import Header from '../header';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <GameBackground>
                <Header />
            </GameBackground>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});