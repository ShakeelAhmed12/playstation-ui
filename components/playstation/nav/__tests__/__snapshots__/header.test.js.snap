// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header__StyledHeader-sc-7cu7g7-0 kxGCHO"
>
  <section
    className="header__StyledNav-sc-7cu7g7-1 lbljKS"
  >
    <a
      className="active"
      href="#"
    >
      Games
    </a>
    <a
      href="#"
    >
      Media
    </a>
  </section>
  <section
    className="header__StyledNav-sc-7cu7g7-1 lbljKS settings"
  >
    <span
      className="icon search"
      onClick={[Function]}
    >
      <svg
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          fill="white"
          id="magnifier"
        />
      </svg>
    </span>
    <span
      className="icon"
    >
      <svg
        enableBackground="new 0 0 24 24"
        fill="white"
        height="48px"
        viewBox="0 0 24 24"
        width="48px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0,0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19.14,12.94c0.04-0.3,0.06-0.61,0.06-0.94c0-0.32-0.02-0.64-0.07-0.94l2.03-1.58c0.18-0.14,0.23-0.41,0.12-0.61 l-1.92-3.32c-0.12-0.22-0.37-0.29-0.59-0.22l-2.39,0.96c-0.5-0.38-1.03-0.7-1.62-0.94L14.4,2.81c-0.04-0.24-0.24-0.41-0.48-0.41 h-3.84c-0.24,0-0.43,0.17-0.47,0.41L9.25,5.35C8.66,5.59,8.12,5.92,7.63,6.29L5.24,5.33c-0.22-0.08-0.47,0-0.59,0.22L2.74,8.87 C2.62,9.08,2.66,9.34,2.86,9.48l2.03,1.58C4.84,11.36,4.8,11.69,4.8,12s0.02,0.64,0.07,0.94l-2.03,1.58 c-0.18,0.14-0.23,0.41-0.12,0.61l1.92,3.32c0.12,0.22,0.37,0.29,0.59,0.22l2.39-0.96c0.5,0.38,1.03,0.7,1.62,0.94l0.36,2.54 c0.05,0.24,0.24,0.41,0.48,0.41h3.84c0.24,0,0.44-0.17,0.47-0.41l0.36-2.54c0.59-0.24,1.13-0.56,1.62-0.94l2.39,0.96 c0.22,0.08,0.47,0,0.59-0.22l1.92-3.32c0.12-0.22,0.07-0.47-0.12-0.61L19.14,12.94z M12,15.6c-1.98,0-3.6-1.62-3.6-3.6 s1.62-3.6,3.6-3.6s3.6,1.62,3.6,3.6S13.98,15.6,12,15.6z"
          />
        </g>
      </svg>
    </span>
    <span
      className="profile"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="User profile avatar"
          decoding="async"
          src="/_next/image?url=%2Fplaystation%2Fprofile.jpg&w=1200&q=75"
          srcSet="/_next/image?url=%2Fplaystation%2Fprofile.jpg&w=320&q=75 320w, /_next/image?url=%2Fplaystation%2Fprofile.jpg&w=420&q=75 420w, /_next/image?url=%2Fplaystation%2Fprofile.jpg&w=768&q=75 768w, /_next/image?url=%2Fplaystation%2Fprofile.jpg&w=1024&q=75 1024w, /_next/image?url=%2Fplaystation%2Fprofile.jpg&w=1200&q=75 1200w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "visible",
              "width": 0,
            }
          }
        />
      </div>
    </span>
    <span
      className="time"
    >
      
    </span>
  </section>
</header>
`;
